### PR TITLE
Add gource

### PR DESCRIPTION
**Pull request by koppor**
[Gource](https://gource.io/) nicely renders the history of our development

See for an example: https://www.youtube.com/watch?v=cFuZnzq97sQ

Refs https://github.com/JabRef/www.jabref.org/issues/83

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: .github/workflows/gource.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  action:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: 'Gource Action'
+        uses: nbprojekt/gource-action@v1.0.1
```
Comment (NBprojekt) [inline .github/workflows/gource.yml:22]: `v1` Tag can be used now
```suggestion
        uses: nbprojekt/gource-action@v1
```

**Comment by Siedlerchr** [comment]
Hm, but I don't see any file yet?

**Comment by NBprojekt** [comment]
The output file is stored in the [artefact](https://github.com/JabRef/jabref/actions/runs/429218588)  
_([download](https://github.com/JabRef/jabref/suites/1689046165/artifacts/31680459))_

If you want to add the file to the repo, you'll need to do it in your own workflow. I could also give an example of how to do this.

**Comment by tobiasdiez** [comment]
It's a nice gimmick, but what are we going to do with the produced movie?

**Comment by Siedlerchr** [comment]
We should upload it to YouTube.

**Comment by koppor** [comment]
> We should upload it to YouTube. 

That's the plan. Have to inspect the length of the video. Maybe add a monthly video and an updated total one covering the whole history 😅

**Comment by Siedlerchr** [comment]
Possible scripts for uploadiing: https://github.com/tokland/youtube-upload
https://github.com/TheMengzor/youtube-upload 
Refs: https://developers.google.com/youtube/v3/guides/uploading_a_video

**Comment by koppor** [comment]
Automatic upload not possible.

1. Server-to-Server authorization not availble for youtube. See last comment in 0 point answer: https://stackoverflow.com/q/29242851/873282
2. Requested OAuth token has a live time for approx 2 hours. Then a manual browser usage is required.

**Comment by Siedlerchr** [comment]
Action fails:
>mv: cannot move './gource/gource.mp4' to './gource-videos/jabref-.mp4': Permission denied

**Comment by tobiasdiez** [comment]
I would propose to only run this workflow via a manual trigger, that can be run from time to time (maybe after each release)?

**Comment by koppor** [comment]
I am trying to reduce the manual steps I have to do.

I can also read error messages.

Sorry that I spend the days between the years differently.

Would you guys mind to focus on other issues and PRs?

**Comment by koppor** [comment]
And sorry for not converting it to draft

**Comment by tobiasdiez** [comment]
xkcd

Sorry, couldn't resist ;-)

**Comment by koppor** [comment]
DevCall: We do not upload video automatically, we just generate it. If that works "quickly" in this PR, we go ahead. Otherwise, we won't have any cool gource videos ^^.

**Comment by Siedlerchr** [comment]
https://crontab.guru/#15_3_1_1,4,7,10_*

“At 03:15 on day-of-month 1 in January, April, July, and October.”

**Comment by calixtus** [comment]
Maybe open an issue "upload to youtube" with a link to the created artifact?

**Comment by koppor** [comment]
Current output:

grafik

grafik

- Q1 is ca. 30 seconds
- complete is ca. 7 minutes

Will try to change the parameter so that the quarter report will also approx. 5 minutes.

**Comment by koppor** [comment]
> “At 03:15 on day-of-month 1 in January, April, July, and October.”

The good thing is that GitHub UI provides a tooltip:

grafik

**Comment by koppor** [comment]
3:30 now. LGTM. I just merge. We can IMHO optimize as soon as we have done the other important things ^^.

**Pull request merged**